OpenHumans/open-humans
# Datafile Datatypes

## Description
This pull request adds the concept of Datatypes to Open Humans.  It provides a mechanism for site users to add new datatypes to the project.  This mechanism is available to all Open Humans members, though it is only linked to the from the Select Datatypes View, which in turn is only available to Project Coordinators.  In turn, projects immediately after creation are sent to this Select Datatypes View, where they can choose the datatypes they will be using with their project, as well as add new datatypes.

This change also enforces that data_files must have one or more datatypes.

## Related Issue
#981 

## Testing
Added to the automated testing

Manually tested:
Create a project, selected a set of datatypes, added a new datatype
Checked the database directly to ensure datatypes were actually saved
Attempted file upload without specifying a datatype, noted that it failed
Attempted file upload with incorrect datatypes, noted it failed
File upload with full set of project datatypes, succeeded
File upload with subset of project datatypes also succeeded.
